test/jdk/java/foreign/enablenativeaccess/panama_module/org/openjdk/foreigntest/libLinkerInvokerModule.cpp

```diff
@@ -52,8 +55,12 @@ extern "C" {
         context.desc = env->NewGlobalRef(desc);
         context.opts = env->NewGlobalRef(opts);
         run_in_new_thread_and_join(call, &context);
+        if (context.exception != nullptr) {
+            env->Throw(context.exception); // transfer exception to this thread
+        }
```

Comment:
Since we invoke the target method on a different thread, the exception wasn't being propagated properly, and as a result, the exit code was just 0, even when we had an error occur.

I've added the exception transfer here.